### PR TITLE
added "**kwargs" to "predict"

### PR DESCRIPTION
**Pull request by zzheng93**

### CODE REVIEW

**Review by sonichi — COMMENTED** [review]
There are a few more places in this file to add `**kwargs` I believe. @liususan091219 could you review the PR and follow up since you introduced `**kwargs` in the `predict()` function?

**Comment by zzheng93** [comment]
> There are a few more places in this file to add `**kwargs` I believe. @liususan091219 could you review the PR and follow up since you introduced `**kwargs` in the `predict()` function?

Hi @sonichi @liususan091219 , I did see more places, but I wasn't sure if they needed to be corrected.

**Comment by zzheng93** [comment]
> > > There are a few more places in this file to add `**kwargs` I believe. @liususan091219 could you review the PR and follow up since you introduced `**kwargs` in the `predict()` function?
> > 
> > 
> > Hi @sonichi @liususan091219 , I did see more places, but I wasn't sure if they needed to be corrected.
> 
> Thanks for your help @zzheng93! Basically you need to add **kwargs to all the places of model.predict inside of the predict function inside model.py (simply control+F ".predict" and add the **kwargs).
> 
> Also, to fix the current pytest error, please change the .get in the following line to .pop:
> 
> `https://github.com/microsoft/FLAML/blob/09eaf0f1b5231bdbe3a1e93476f9b0494c77d9dc/flaml/model.py#L315`

Hi @liususan091219 , sure, I'll do it!

File: flaml/model.py [hunk 1]
```diff
@@ -858,7 +858,7 @@ def predict_proba(self, X, **pred_kwargs):
         test_dataset = Dataset.from_pandas(X_test)
 
         new_trainer = self._init_model_for_predict()
-        predictions = new_trainer.predict(test_dataset)
+        predictions = new_trainer.predict(test_dataset, **pred_kwargs)
```
Comment (liususan091219) [inline flaml/model.py:861]:
```suggestion
        predictions = new_trainer.predict(test_dataset)

```

File: flaml/model.py [hunk 2]
```diff
@@ -890,10 +890,10 @@ def predict(self, X, **pred_kwargs):
         new_trainer = self._init_model_for_predict()
 
         if self._task not in NLG_TASKS:
-            predictions = new_trainer.predict(test_dataset)
+            predictions = new_trainer.predict(test_dataset, **pred_kwargs)
```
Comment (liususan091219) [inline flaml/model.py:893]:
```suggestion
            predictions = new_trainer.predict(test_dataset)

```

File: flaml/model.py [hunk 3]
```diff
@@ -890,10 +890,10 @@ def predict(self, X, **pred_kwargs):
         new_trainer = self._init_model_for_predict()
 
         if self._task not in NLG_TASKS:
-            predictions = new_trainer.predict(test_dataset)
+            predictions = new_trainer.predict(test_dataset, **pred_kwargs)
         else:
             predictions = new_trainer.predict(
-                test_dataset,
+                test_dataset, **pred_kwargs,
```
Comment (liususan091219) [inline flaml/model.py:896]:
```suggestion
                test_dataset,

```

**Review by liususan091219 — CHANGES_REQUESTED** [review]

**Comment by zzheng93** [comment]
Hi @liususan091219, I have just committed and merged. Let me know if it works. Thanks!

**Review by liususan091219 — APPROVED** [review]

**Comment by liususan091219** [comment]
> Hi @liususan091219, I have just committed and merged. Let me know if it works. Thanks!

Looks good. Thank you!

**Pull request merged**